core/src/services/cloudflare_kv/backend.rs

```diff
@@ -0,0 +1,387 @@
+                let body = resp.into_body().bytes().await?;
+                Ok(Some(body.into()))
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn set(&self, path: &str, value: &[u8]) -> Result<()> {
+        let url = format!("{}/values/{}", self.url_prefix, path);
+        let req = Request::put(&url);
+        let multipart = Multipart::new();
+        let multipart = multipart
+            .part(FormDataPart::new("metadata").content(serde_json::Value::Null.to_string()))
+            .part(FormDataPart::new("value").content(value.to_vec()));
+        let mut req = multipart.apply(req)?;
+        req = self.sign(req)?;
+        let resp = self.client.send(req).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => Ok(()),
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn delete(&self, path: &str) -> Result<()> {
+        let url = format!("{}/values/{}", self.url_prefix, path);
+        let mut req = Request::delete(&url);
+        req = req.header(header::CONTENT_TYPE, "application/json");
+        let mut req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        req = self.sign(req)?;
+        let resp = self.client.send(req).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => Ok(()),
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+
+    async fn scan(&self, path: &str) -> Result<Vec<String>> {
+        let mut url = format!("{}/keys", self.url_prefix);
+        if !path.is_empty() {
+            url = format!("{}?prefix={}", url, path);
+        }
+        let mut req = Request::get(&url);
+        req = req.header(header::CONTENT_TYPE, "application/json");
+        let mut req = req
+            .body(AsyncBody::Empty)
+            .map_err(new_request_build_error)?;
+        req = self.sign(req)?;
+        let resp = self.client.send(req).await?;
+        let status = resp.status();
+        match status {
+            StatusCode::OK => {
+                let body = resp.into_body().bytes().await?;
+                let response: CfKvScanResponse = serde_json::from_slice(&body).map_err(|e| {
+                    Error::new(
+                        crate::ErrorKind::Unexpected,
+                        &format!("failed to parse error response: {}", e),
+                    )
+                })?;
+                Ok(response.result.into_iter().map(|r| r.name).collect())
+            }
+            _ => Err(parse_error(resp).await?),
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct CfKvResponse {
+    pub(crate) errors: Vec<CfKvError>,
+    // pub(crate) messages: Vec<CfKvMessage>,
+    // pub(crate) result: serde_json::Value,
+    // pub(crate) success: bool,
+}
+
+#[derive(Debug, Deserialize)]
+pub(crate) struct CfKvScanResponse {
+    // errors: Vec<CfKvError>,
```

Comment:
Please remove those unused fields.

How about adding links to this struct like https://developers.cloudflare.com/api/operations/workers-kv-namespace-list-a-namespace'-s-keys with comments about possible fields?